examples/crystal.py

```diff
@@ -327,11 +346,28 @@ def simulate_crystal_with_gaussian_beam(
     # beam, and then solving for the eigenmodes.
     # TODO: replace paraxial Gaussian with a more rigorous expression.
 
-    def _paraxial_gaussian_field_fn(x, y, z):
+    def _paraxial_gaussian_field_fn(
+        x,
+        y,
+        z,
+    ):
         # Returns the fields of a z-propagating, x-polarized Gaussian beam.
         # See https://en.wikipedia.org/wiki/Gaussian_beam
-        k = 2 * jnp.pi / wavelength
-        z_r = jnp.pi * beam_waist**2 * jnp.sqrt(permittivity_ambient) / wavelength
+
+        # Adjust array dimensions for proper batching
+        batch_axes = (WAVELENGTH_AXIS, WAVELENGTH_AXIS + 1, WAVELENGTH_AXIS + 2)
+        x = jnp.expand_dims(x, axis=batch_axes)
```

Comment:
I didn't go through this carefully, but can you confirm that you actually need to expand the dims of `(x, y, z)` here? It seems like these will be shape `(nx, ny)`, and so by adding the extra dims to `wavelengths_padded` we already achieve compatibility.